watertap/unit_models/tests/test_crystallizer.py

```diff
@@ -101,6 +101,52 @@ def Crystallizer_frame(self):
 
         return m
 
+    @pytest.fixture(scope="class")
+    def Crystallizer_frame_2(self):
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(default={"dynamic": False})
+
+        m.fs.properties = props.NaClParameterBlock()
+
+        m.fs.unit = Crystallization(
+            default={
+                "property_package": m.fs.properties,
+            }
+        )
+
+        # fully specify system
+        feed_flow_mass = 1
+        feed_mass_frac_NaCl = 0.2126
+        feed_mass_frac_H2O = 1 - feed_mass_frac_NaCl
+        feed_pressure = 101325
+        feed_temperature = 273.15 + 20
+        eps = 1e-6
+        crystallizer_temperature = 273.15 + 55
+        crystallizer_yield = 0.40
+
+        # Fully define feed
+        m.fs.unit.inlet.flow_mass_phase_comp[0, "Liq", "NaCl"].fix(
+            feed_flow_mass * feed_mass_frac_NaCl
+        )
+        m.fs.unit.inlet.flow_mass_phase_comp[0, "Liq", "H2O"].fix(
+            feed_flow_mass * feed_mass_frac_H2O
+        )
+        m.fs.unit.inlet.flow_mass_phase_comp[0, "Sol", "NaCl"].fix(eps)
+        m.fs.unit.inlet.flow_mass_phase_comp[0, "Vap", "H2O"].fix(eps)
+        m.fs.unit.inlet.pressure[0].fix(feed_pressure)
+        m.fs.unit.inlet.temperature[0].fix(feed_temperature)
+
+        # Define operating conditions
+        m.fs.unit.temperature_operating.fix(crystallizer_temperature)
+        m.fs.unit.crystallization_yield["NaCl"].fix(crystallizer_yield)
+
+        # Fix growth rate, crystal length and Sounders brown constant to default values
+        m.fs.unit.crystal_growth_rate.fix()
+        m.fs.unit.souders_brown_constant.fix()
+        m.fs.unit.crystal_median_length.fix()
+
```

Comment:
I would check that `assert_units_consistent` holds true, once after building the model, and another time after adding costing.

Reply:
Done